logger/drain/factory.go

```diff
@@ -15,19 +14,14 @@ func NewDrain(drainURL string) (LogDrain, error) {
 		// nil means no drain-- which is valid
 		return nil, nil
 	}
-	if strings.HasPrefix(drainURL, "udp://") || strings.HasPrefix(drainURL, "syslog://") {
-		drain, err := udp.NewDrain(drainURL)
-		if err != nil {
-			return nil, err
-		}
-		return drain, nil
-	}
-	if strings.HasPrefix(drainURL, "tcp://") {
-		drain, err := tcp.NewDrain(drainURL)
+	// Any of these three can use the same drain implementation
+	if strings.HasPrefix(drainURL, "udp://") || strings.HasPrefix(drainURL, "syslog://") || strings.HasPrefix(drainURL, "tcp://") {
```

Comment:
I like that you've absorbed the drain 'types' into the `simple` package. I wonder if it wouldn't be even better to absorb the conditional prefix string logic into the NewDrain 'constructor' as well. And in fact, the same for future implementations, e.g., TLS/TCP and HTTP(s) as mentioned below.